km/km_fork.c

```diff
@@ -60,6 +60,26 @@ static km_fork_state_t km_fork_state = {
     .fork_in_progress = 0,
 };
 
+/*
+ * If this process has successfully fork()'ed, we don't allow snapshots to be taken.
+ * Snapshots must preserve unread data buffered in pipes and socketpairs and then write
+ * that data back into the other end of the pipe/socketpair.  If a fork
+ * has happened snapshot can't tell if a pipe or socketpair with a single end in this process
+ * has no other end, or the other end is in another process.
+ * If the other end is in another process, we can't snapshot buffered data.
+ * This counter is used by snapshot to disable snapshots after a fork has be done.
+ */
+unsigned int km_fork_count = 0;
```

Comment:
We might want to count children that exited, so even if there was fork but the child is gone we allow the snapshot.

Reply:
What about children of children etc.

It might get to the point that we should just analyze processes that are descendants of the process being snapshotted.
If there are currently any descendants don't allow the snapshot.